libpod/container_api.go

```diff
@@ -376,14 +376,18 @@ func (c *Container) Exec(tty, privileged bool, env, cmd []string, user string) e
 	logrus.Debugf("Successfully started exec session %s in container %s", sessionID, c.ID())
 
 	// Unlock so other processes can use the container
-	c.lock.Unlock()
-	locked = false
+	if !c.locked {
```

Comment:
should this be:
```
if c.locked
```
?  Otherwise aren't you trying to unlock an unlocked container?

Reply:
`c.locked` is a variable that indicates whether we are in a batched operation (IE, individual commands should not do any locking themselves, the container has been locked externally)